substrate/frame/revive/src/wasm/runtime.rs

```diff
@@ -1091,8 +1094,11 @@ impl<'a, E: Ext, M: ?Sized + Memory<E::T>> Runtime<'a, E, M> {
 		salt_ptr: u32,
 	) -> Result<ReturnErrorCode, TrapReason> {
 		self.charge_gas(RuntimeCosts::Instantiate { input_data_len })?;
-		let deposit_limit: U256 =
-			if deposit_ptr == SENTINEL { U256::zero() } else { memory.read_u256(deposit_ptr)? };
+		let deposit_limit: U256 = if deposit_ptr == SENTINEL {
+			U256::from(u64::MAX)
+		} else {
+			memory.read_u256(deposit_ptr)?
+		};
```

Comment:
```suggestion
		memory.read_u256(deposit_ptr)?
```

Reply:
This raises `OutOfBounds` errors in many tests (although it replaced the last 2 tests' failure `ContractTrapped` failure, which is progress).

Do you have an idea of what might be causing this?

Reply:
The test fixtures need to be adapted to always pass a valid pointer. `SENTINEL` is used to indicate a `None` so to say.

Reply:
For example, in [`substrate/frame/revive/fixtures/contracts/call.rs`](https://github.com/paritytech/polkadot-sdk/blob/7531f72eee895357c07c33bad067919c0302117e/substrate/frame/revive/fixtures/contracts/call.rs#L43), the deposit limit is `None` - this is was we want, right

Reply:
Yes. You need to change the wrapper in `pallet-fixtures-uapi`. It no longer takes an `Option` since the value is no longer optional. You then pass in `U256::MAX` where `None` was passes before.

Reply:
Where there is `U256::MAX`, do you mean `&[u8::MAX; 32]`, since in `trait HostFn` one has `deposit_limit: &[u8; 32],`?

Also, won't we want something like `&[u8::MAX; 8].concat([0u8; 24]`? Using `&[u8::MAX; 32]` will raise `Error::BalanceConversionFailed`.